app/controllers/api/v2/courses_controller.rb

```diff
@@ -65,8 +65,19 @@ def index
       end
 
       def show
-        # https://github.com/jsonapi-rb/jsonapi-rails/issues/113
-        render jsonapi: @course, include: params[:include], class: CourseSerializersService.new.execute
+        json_data = JSONAPI::Serializable::Renderer.new.render(
+          @course,
+          class: CourseSerializersService.new.execute,
+          include: params[:include],
+        )
+
+        unless @current_user.admin?
```

Comment:
this feels like it should be handled in the serializer. You can expose the user information to the serializer via `expose:` and deal with the logic in there?

Reply:
ie https://github.com/DFE-Digital/teacher-training-api/blob/master/app/controllers/api/public/v1/providers/courses/locations_controller.rb#L10

Reply:
discussed doing this alternatively offline, but potentially in a future PR